src/settings-ui/Settings.UI.Library/SettingsBackupAndRestoreUtils.cs

```diff
@@ -0,0 +1,893 @@
+                    {
+                        property.WriteTo(jsonWriter);
+                    }
+                }
+
+                jsonWriter.WriteEndObject();
+            }
+
+            private static void MergeArrays(Utf8JsonWriter jsonWriter, JsonElement root1, JsonElement root2, bool allowDupes)
+            {
+                // just does one level!!!
+                jsonWriter.WriteStartArray();
+
+                if (allowDupes)
+                {
+                    // Write all the elements from both JSON arrays
+                    foreach (JsonElement element in root1.EnumerateArray())
+                    {
+                        element.WriteTo(jsonWriter);
+                    }
+
+                    foreach (JsonElement element in root2.EnumerateArray())
+                    {
+                        element.WriteTo(jsonWriter);
+                    }
+                }
+                else
+                {
+                    var arrayItems = new HashSet<string>();
+                    foreach (JsonElement element in root1.EnumerateArray())
+                    {
+                        element.WriteTo(jsonWriter);
+                        arrayItems.Add(element.ToString());
+                    }
+
+                    foreach (JsonElement element in root2.EnumerateArray())
+                    {
+                        if (!arrayItems.Contains(element.ToString()))
+                        {
+                            element.WriteTo(jsonWriter);
+                        }
+                    }
+                }
+
+                jsonWriter.WriteEndArray();
+            }
+        }
+
+        private static bool TryCreateDirectory(string path)
+        {
+            try
+            {
+                if (!Directory.Exists(path))
+                {
+                    Directory.CreateDirectory(path);
+                    return true;
+                }
+            }
+            catch (Exception ex3)
+            {
+                Logger.LogError($"There was an error in TryCreateDirectory {path}: {ex3.Message}", ex3);
+                return false;
+            }
+
+            return true;
+        }
+
+        private static bool TryDeleteDirectory(string path)
+        {
+            try
+            {
+                if (Directory.Exists(path))
+                {
+                    Directory.Delete(path, true);
+                    return true;
+                }
+            }
+            catch (Exception ex3)
+            {
+                Logger.LogError($"There was an error in TryDeleteDirectory {path}: {ex3.Message}", ex3);
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Method <c>SetRegSettingsBackupAndRestoreItem</c> helper method to write to the registry.
+        /// </summary>
+        public static void SetRegSettingsBackupAndRestoreItem(string itemName, string itemValue)
```

Comment:
I don't think we do because it figures this out by itself (https://learn.microsoft.com/en-us/dotnet/api/microsoft.win32.registrykey.setvalue?view=net-6.0)